### ISSUE TITLE
Bug: SSH private key permissions appear to reset with new JupyterHub instance

### ISSUE THREAD

**Issue by edasmalchi** [cal-itp/data-infra#873]
**Describe the bug**
After setting up SSH for GitHub on the JupyterHub, it's necessary to set permissions for the private key to 600 or the ssh client won't work (see closed issue [792](https://github.com/cal-itp/data-infra/issues/792)).

While that advice is now in the docs, it seems like the permissions actually reset between JupyterHub instances, requiring them to be reset every time instead of set up once and done.

**To Reproduce**
1. Set up SSH on JupyterHub
2. Wait a while
3. Try to use git again

**Expected behavior**
Not having to reset permissions -- can we make them persistent between instances of a user's JupyterHub?

**Additional context**

**Comment by lauriemerrell** [comment]
I am running into this issue as well, exactly as described. When I start a new JupyterHub session, I get the error that the permissions are too lax on the SSH key, despite having previously tightened the access on that file using the `chmod 600 /home/jovyan/.ssh/id_ed25519` command as noted in [the docs](https://docs.calitp.org/data-infra/analytics_tools/saving_code.html).

**Comment by machow** [comment]
Currently people support using ssh through the following steps:

* create an `~/.ssh` directory
* set permissions in the directory (or on specific files there); e.g. to 600

I wonder if the chmod settings don't get saved, but we can run a post startup hook to set them each time we start a container in this case?

**Comment by lottspot** [comment]
This could probably be fixed by adding to the `postStart` hook @atvaccaro created in 2263521b66d20d32579f76a36c1d35d4ec9a3d51

**Comment by atvaccaro** [comment]
Yeah I think we can close this. I'd rather avoid doing something in `postStart` since people will have to create the key each time anyways. Happy to re-open and figure something out if needed later.

**Comment by atvaccaro** [comment]
Oh just kidding I actually did implement this https://github.com/cal-itp/data-infra/commit/6ab5c7e510fcef2d8457566b6c07297bafa36d57#diff-cdf8a5d2e5b42697d69c4ef74cbe5a8f2b6e7db97145545ea7270e47f1bc3699R5
